CIP-0033/README.md

```diff
@@ -0,0 +1,117 @@
+---
+CIP: 33
+Title: Reference scripts
+Authors: Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Comments-Summary: No comments
+Comments-URI: 
+Status: Draft
+Type: Standards Track
+Created: 2021-11-29
+License: CC-BY-4.0
+Requires: CIP-31
+---
+
+# Reference scripts
+
+## Abstract
+
+We propose to allow scripts ("reference scripts") to be attached to outputs, and to allow reference scripts to be used to satisfy script requirements during validation, rather than requiring the spending transaction to do so.
+This will allow transactions using common scripts to be much smaller.
```

Comment:
Sorry for probably stupid question. Do you think that this idea could  be extended to something like content addressable imports in plutus core - something similar to [`unison`](https://www.unisonweb.org/docs/tour#%F0%9F%A7%A0-the-big-technical-idea) or `nix` artifact build strategy.

Reply:
No, that's drastically different and more difficult. This proposal is entirely external to the language: it just treats scripts as black boxes and lets you get those boxes from other places. Something that actually intruded into the language itself would be massively more complicated.